docs/design/velero-krm-enhancement.md

```diff
@@ -0,0 +1,103 @@
+---
+title: velero-openshiftkrm-plugin design
+authors:
+  - "@ChaitanyaBhangale"
+reviewers:
+  - "@jwmatthews" 
+  - "@shubham-pampattiwar"
+  - "@dylan-murray"
+  - "@tiger-kaovilai"
+approvers:
+  - "@jwmatthews" 
+  - "@dylan-murray"
+  - "@shubham-pampattiwar"
+creation-date: 2022-06-21
+last-updated: 2022-06-21
+status: 
+  - "N/A"
+see-also:
+  - "N/A"  
+replaces:
+  - "N/A"
+superseded-by:
+  - "N/A"
+---
+
+# Velero KRM Plugin Design
+
+### Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] User-facing documentation is created
+
+### Open Questions 
+-  Can changes be made in exising codebase as opposed to refactoring the whole code? 
+-  Whether adding KRM (Kubernetes Resource Management) functions is actually useful or will it affect the performance of current plugin in a negative manner?
+
+
+### Summary
+
+Currently Velero plugin framework is inefficient. Velero server executes 1 gRPC call per plugin, meaning 1 gRPC call per resource transformation.  In order to overcome this we introduce a new plugin type that will allow Velero to leverage KRM functions for resource transformation and additional item gathering so that we can drastically improve the plugin framework performance by processing N items with a single gRPC call.
```

Comment:
this sounds like a motivation, or a reason to do the work, I think the summary should be a high level of what you would change.